### ISSUE TITLE
iCubGenova09 (iRonCub3) S/N:000 – Right hip roll in hardware fault at startup

### ISSUE THREAD

**Issue by gabrielenava** [robotology/icub-tech-support#1652]
### Robot Name 🤖

iCubGenova09 (iRonCub3) S/N:000

### Request/Failure description

The right hip roll goes in hardware fault at startup. The joint is not recoverable even later on from motorgui, pressing idle or calib. the encoder measurement is strange (110 degrees instead of zero).

I will add pictures and logger while we try again the startup phase for debugging the problem.

### Detailed context

cc @davidegorbani @vpunithreddy

**Comment by davidegorbani** [comment]
Here is the picture of the `yarpmotorgui` related to the `r_hip_roll`, the file of the `yarplogger` when we started the robot and a picture of the cable of the right leg.

![Screenshot from 2023-10-12 11-48-34](https://github.com/robotology/icub-tech-support/assets/101326942/fa0dd0ac-ffe2-471c-a609-d6771d80bafc)


![MicrosoftTeams-image](https://github.com/robotology/icub-tech-support/assets/101326942/0f3ff8c5-6718-4ef9-865e-4217d9f8f22b)


[right_leg_overcurrent.zip](https://github.com/robotology/icub-tech-support/files/12879947/right_leg_overcurrent.zip)

**Comment by DanielePucci** [comment]
Probably it would be better to add the glue when fixing the connector @AntonioConsilvio @sgiraz

**Comment by AntonioConsilvio** [comment]
Hi @gabrielenava, the problem was caused by failure to read the motor encoder (LCORE) connected to 2FOC 0B10.

The sensor was not reading because the connector on the 2FOC board stopped working.

So I replaced the 2FOC with a new one, and the robot was tested and it works properly now!

cc @sgiraz

**Comment by gabrielenava** [comment]
>So I replaced the 2FOC with a new one, and the robot was tested and it works properly now!

thank you @AntonioConsilvio, but I would wait before closing this issue: I have heard that @HosameldinMohamed tested the robot yesterday and the jip roll was going in hardware fault. when moved. 

Waiting for @HosameldinMohamed to have more details and confirm that there is still this problem!

**Comment by gabrielenava** [comment]
EDIT: I see that @HosameldinMohamed just opened https://github.com/robotology/icub-tech-support/issues/1679 for tracking this problem. Being the same joint, I think the problem is related to the one of this issue, but since a new issue has been opened we can close this one and continue the analysis in the other issue.

**Comment by HosameldinMohamed** [comment]
Oh sorry I just saw this issue!

**Comment by sgiraz** [comment]
> Oh sorry I just saw this issue!

just for reference: https://github.com/robotology/icub-tech-support/issues/1679

**Comment by davidegorbani** [comment]
Here is the plot of the current of the right hip roll when moving it of about 20 degrees.

![currentPlot](https://github.com/robotology/icub-tech-support/assets/101326942/ecfb8fb7-1a09-4e1e-82ca-7d0a6d765ac8)

**Comment by GiacomoVassallo13** [comment]
Hi. The faulted motor was the hip roll of right leg not the waist. 
The faulted motor has been replace wich a new one. 
I attach the photo of the replaced stator:

![Image](https://github.com/robotology/icub-tech-support/assets/70649516/03fe8742-3f80-4925-9022-b492e0984c70)

The optical disc was also replaced as it was worn ,and the correct reading was verified with the bench test.

Unfortunately, I would like to point out that the assembly is missing some components: in particular the front edges of the sprockets and a  piece of the belt tensioner.

![AAAAAAAAAAAAA](https://github.com/icub-tech-iit/proto/assets/70649516/65b80a80-d8fb-4b3a-a6a0-52aa32084fd6)

This components will be added the as they become available.

The motor group was reassembled on the robot. @AntonioAzocar will take care of the electronic wiring of the new engine.

I attend the feedback. 


cc: @Fabrizio69 @sgiraz @AntonioConsilvio

**Comment by HosameldinMohamed** [comment]
Today @AntonioConsilvio checked the AMO sensor using https://icub-tech-iit.github.io/procedures/tp-amo-calibration/#magnetic-ring-or-amo-board-incorrectly-mounted (you may add details if you prefer), and it seemed OK!

Then he performed a stator-rotor calibration https://icub-tech-iit.github.io/documentation/icub_robot_calibration/icub3-stator-rotor-autocalibration/

Now the joint is working as expected. We moved the repaired joint (hip roll) and it moves without a fault. Also, comparing with the left one, the current values seem similar, However, the estimated torque values are different!

**Comment by AntonioConsilvio** [comment]
Hi! Thanks @HosameldinMohamed for the summary! 

Regarding the AMO, what I did was to perform the calibration without installing on the RAM the results. I did this just to see if the magnetic ring was mounted correctly (see an example in the [documentation](https://icub-tech-iit.github.io/procedures/tp-amo-calibration/#magnetic-ring-or-amo-board-incorrectly-mounted)).

The robot is working correctly now!

> [!Note]
> I would also like to report an anomaly, in fact during the rotor-stator calibration, I saw that the old offset value present in the hip roll joint was 0 (inside the mechanicals file). 
That offset comes from the rotor-stator calibration, and as far as I know, it can't be 0, so its presence in the configuration files is strange.
> 
> Maybe that negatively affected the motor and caused it to burn out.

However, since the problem is solved, I'll close to the issue!

cc @sgiraz @gabrielenava
